src/proc/color-formats-converter.cpp

```diff
@@ -354,6 +354,195 @@ namespace librealsense
 #endif
     }
 
+    template<rs2_format FORMAT>
+    void parse_one_line(const byte* y_one_line, const byte* uv_one_line, uint8_t** dst, int width)
+    {
+        // building 16 pixels at each iteration 
```

Comment:
Do we understand why a batch of 16 is used?

Reply:
This is the same code as for YUYV parser.
Maybe we just assume that the resolutions can all be divided by 16. Moreover, this is power of 2, so it is easier for the memory.
Got no other reason - maybe we can try to do it in one iteration, and with arrays of length width. Let's talk about it.

Reply:
@ev-mp Do you know maybe?